cirq/circuits/qasm_output.py

```diff
@@ -25,6 +25,7 @@
 from cirq import ops, linalg, protocols, value
 
 
+@value.value_equality(unhashable=True, approximate=True)
```

Comment:
Why unhashable?

Reply:
yeah, it is not strictly necessary, I can fix it in the next one. 
 
I am still unsure what is the use of a __hash__ function in the approximate equality case? In my understanding, checking equality on these classes is not useful due to the floating point nature of the underlying values. And having the default `_value_equality_hash` implementation is more expensive than a `None`...

Reply:
approximate=True does not change the behavior of `__eq__`, it adds a method `_approx_eq_`.